src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -600,25 +600,27 @@ public static void Run(ServiceBase[] services)
             if (services == null || services.Length == 0)
                 throw new ArgumentException(SR.NoServices);
 
-            IntPtr entriesPointer = Marshal.AllocHGlobal((IntPtr)((services.Length + 1) * Marshal.SizeOf(typeof(SERVICE_TABLE_ENTRY))));
+            int sizeOfSERVICE_TABLE_ENTRY = Marshal.SizeOf<SERVICE_TABLE_ENTRY>();            
+
+            IntPtr entriesPointer = Marshal.AllocHGlobal(checked((services.Length + 1) * sizeOfSERVICE_TABLE_ENTRY));
             SERVICE_TABLE_ENTRY[] entries = new SERVICE_TABLE_ENTRY[services.Length];
             bool multipleServices = services.Length > 1;
-            IntPtr structPtr = (IntPtr)0;
+            IntPtr structPtr;
 
             for (int index = 0; index < services.Length; ++index)
             {
                 services[index].Initialize(multipleServices);
                 entries[index] = services[index].GetEntry();
-                structPtr = (IntPtr)((long)entriesPointer + Marshal.SizeOf(typeof(SERVICE_TABLE_ENTRY)) * index);
-                Marshal.StructureToPtr(entries[index], structPtr, true);
+                structPtr = entriesPointer + sizeOfSERVICE_TABLE_ENTRY * index;
+                Marshal.StructureToPtr(entries[index], structPtr, fDeleteOld: false);
             }
 
             SERVICE_TABLE_ENTRY lastEntry = new SERVICE_TABLE_ENTRY();
 
             lastEntry.callback = null;
             lastEntry.name = (IntPtr)0;
-            structPtr = (IntPtr)((long)entriesPointer + Marshal.SizeOf(typeof(SERVICE_TABLE_ENTRY)) * services.Length);
-            Marshal.StructureToPtr(lastEntry, structPtr, true);
+            structPtr = entriesPointer + sizeOfSERVICE_TABLE_ENTRY * services.Length;
+            Marshal.StructureToPtr(lastEntry, structPtr, fDeleteOld: false);
```

Comment:
It does not make sense to pass `fDeleteOld: true` here since the memory is uninitialized.